pkg/build/builder/util_linux.go

```diff
@@ -5,15 +5,28 @@ import (
 	"os"
 
 	"github.com/opencontainers/runc/libcontainer/cgroups"
-
 	s2iapi "github.com/openshift/source-to-image/pkg/api"
 )
 
 // GetCGroupLimits returns a struct populated with cgroup limit values gathered
 // from the local /sys/fs/cgroup filesystem.  Overflow values are set to
 // math.MaxInt64.
 func GetCGroupLimits() (*s2iapi.CGroupLimits, error) {
-	byteLimit, err := readInt64("/sys/fs/cgroup/memory/memory.limit_in_bytes")
+	// see https://git.kernel.org/pub/scm/linux/kernel/git/tj/cgroup.git/tree/Documentation/admin-guide/cgroup-v2.rst
+	// for list of cgroupv2 files to try, but Nalin relayed that examination of the crun and runc code that 'memory.high'
+	// is not used.
+	filesToTry := []string{"/sys/fs/cgroup/memory/memory.limit_in_bytes",
+		"/sys/fs/cgroup/memory.max",
+	}
+	var byteLimit int64
+	var err error
+	for _, filePath := range filesToTry {
```

Comment:
Trying to read cgroup v2 file on a cgroup v1 system, or vice versa, is not very efficient.

Trying to read cgroup v2 file on _any_ error from reading cgroup v1 file is also not good.

I suggest using something like IsCgroupV2UnifiedMode() from https://github.com/opencontainers/runc/blob/960b5fdb5acde1063960a80eb224672faf244e02/libcontainer/cgroups/utils.go#L34

and do this:
```go
file := "/sys/fs/cgroup/memory/memory.limit_in_bytes"
if IsCgroupV2UnifiedMode() {
   file = "/sys/fs/cgroup/memory.max"
}
byteLimit, err := readMaxStringOrInt64(file)
....
```